src/methods.jl

```diff
@@ -10,318 +19,453 @@ derivatives.
 
+_backward_grid(p::Int) = (1 - p):0
 
-## Deprecations
+function _central_grid(p::Int)
+    if isodd(p)
+        return div(1 - p, 2):div(p - 1, 2)
+    else
+        return vcat(div(-p, 2):-1, 1:div(p, 2))
+    end
+end
 
-# Used for keyword argument name deprecations
-function _dep_kwarg(kwargs)
-    for (old, new) in [(:ε, :eps), (:M, :bound)]
-        haskey(kwargs, old) || continue
-        val = kwargs[old]
-        error(
-            "keyword argument `", old, "` should now be passed as `", new, "` upon ",
-            "application of the method. For example:\n    ",
-            "central_fdm(5, 1)(f, x; $new=$val)\n",
-            "not\n    ",
-            "central_fdm(5, 1; $old=$val)(f, x)"
-        )
+_exponentiate_grid(grid::Vector, base::Int=3) = sign.(grid) .* base .^ abs.(grid) ./ base
+
+function _is_symmetric(vec::Vector; centre_zero=false, negate_half=false)
+    n = div(length(vec), 2)
+    half_sign = negate_half ? -1 : 1
+    if isodd(length(vec))
+        centre_zero && vec[n + 1] != 0 && return false
+        return vec[1:n] == half_sign .* reverse(vec[n + 2:end])
+    else
+        return vec[1:n] == half_sign .* reverse(vec[n + 1:end])
     end
 end
 
-function fdm(
-    grid::AbstractVector{<:Real},
-    q::Int,
-    ::Union{Val{true}, Val{false}}=Val(false);
-    kwargs...,
-)
-    error("to use a custom grid, use `Nonstandard(grid, q)` and pass the result to `fdm`")
+function _is_symmetric(m::FiniteDifferenceMethod)
+    grid_symmetric = _is_symmetric(m.grid, centre_zero=true, negate_half=true)
+    coefs_symmetric =_is_symmetric(m.coefs, negate_half=true)
+    return grid_symmetric && coefs_symmetric
 end
 
-for fdmf in (:central_fdm, :backward_fdm, :forward_fdm)
-    @eval function $fdmf(p::Int, q::Int, ::Union{Val{true}, Val{false}}; kwargs...)
-        error(
-            "the `Val` argument should now be passed directly to `fdm` after ",
-            "constructing the method, not to the method constructor itself"
-        )
+"""
+    extrapolate_fdm(
+        m::FiniteDifferenceMethod,
+        f::Function,
+        x::T;
+        factor::Int=1000,
+        kw_args...
+    ) where T<:AbstractFloat
+
+Use Richardson extrapolation to refine a finite difference method. This method uses
+[`estimate_step`](@ref) to determine an appropriate initial step size for
+`Richardson.extrapolate`.
+
+Takes further in keyword arguments for `Richardson.extrapolate`. This method
+automatically sets `power = 2` if `m` is symmetric.
+
+# Arguments
+- `m::FiniteDifferenceMethod`: Finite difference method to estimate the step size for.
+- `f::Function`: Function to evaluate the derivative of.
+- `x::T`: Point to estimate the derivative at.
+
+# Factor
+- `factor::Int=1000`: Factor to amplify the estimated step size by.
+
+# Returns
+- `Tuple{Real, Real}`: Estimate of the derivative and error.
+"""
+function extrapolate_fdm(
+    m::FiniteDifferenceMethod,
+    f::Function,
+    x::T;
+    factor::Int=1000,
```

Comment:
is ther a reason this has to be an integer?

Reply:
Not really. It just has to be something that will be multiplied to the output of `eps(x)`. Should we remove the type?

Reply:
`::Real`?